src/test/kotlin/com/safehill/kclient/api/SafehillApiImplTests.kt

```diff
@@ -1,26 +1,36 @@
 package com.safehill.kclient.api
 
-import com.safehill.kclient.api.dtos.SHAuthResponseDTO
 import com.safehill.kclient.api.dtos.SHAssetOutputDTO
-import com.safehill.kclient.models.*
+import com.safehill.kclient.api.dtos.SHAuthResponseDTO
 import com.safehill.kclient.api.dtos.SHSendCodeToUserRequestDTO
+import com.safehill.kclient.models.SHAssetDescriptor
+import com.safehill.kclient.models.SHAssetQuality
+import com.safehill.kclient.models.SHEncryptedAssetImpl
+import com.safehill.kclient.models.SHEncryptedAssetVersionImpl
+import com.safehill.kclient.models.SHLocalUser
+import com.safehill.kclient.models.SHServerUser
 import com.safehill.kcrypto.models.SHKeyPair
 import com.safehill.kcrypto.models.SHLocalCryptoUser
-import kotlinx.coroutines.*
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.runBlocking
 import org.junit.jupiter.api.Test
 import java.util.Base64
 import java.util.Date
 import kotlin.random.Random
 
 class SafehillApiImplTests {
 
-    private suspend fun createUserOnServer(coroutineScope: CoroutineScope, user: SHLocalUser? = null): SHLocalUser {
+    private suspend fun createUserOnServer(
+        coroutineScope: CoroutineScope,
+        user: SHLocalUser? = null
+    ): SHLocalUser {
         val localUser: SHLocalUser = user ?: run {
             val cryptoUser = SHLocalCryptoUser()
             SHLocalUser(cryptoUser)
         }
 
-        val charPool : List<Char> = ('a'..'z') + ('A'..'Z') + ('0'..'9')
```

Comment:
Do you mind defining carving out username validation and define a new method for it?

Reply:
I am not sure what you mean. Could you elaborate a bit? Do you want me to extract out the function that returns username?